helix-core/src/main/java/org/apache/helix/model/CloudConfig.java

```diff
@@ -0,0 +1,290 @@
+    _record.setSimpleField(CloudConfigProperty.CLOUD_ID.name(), cloudID);
+  }
+
+  /**
+   * Get the CloudID field.
+   * @return CloudID
+   */
+  public String getCloudID() {
+    return _record.getSimpleField(CloudConfigProperty.CLOUD_ID.name());
+  }
+
+  /**
+   * Set the CLOUD_INFO_SOURCE field.
+   * @param cloudInfoSources
+   */
+  public void setCloudInfoSource(List<String> cloudInfoSources) {
+    _record.setListField(CloudConfigProperty.CLOUD_INFO_SOURCE.name(), cloudInfoSources);
+  }
+
+  /**
+   * Get the CLOUD_INFO_SOURCE field.
+   * @return CLOUD_INFO_SOURCE field.
+   */
+  public List<String> getCloudInfoSources() {
+    return _record.getListField(CloudConfigProperty.CLOUD_INFO_SOURCE.name());
+  }
+
+  /**
+   * Set the CLOUD_INFO_PROCESSOR_NAME field.
+   * @param cloudInfoProcessorName
+   */
+  public void setCloudInfoFProcessorName(String cloudInfoProcessorName) {
+    _record.setSimpleField(CloudConfigProperty.CLOUD_INFO_PROCESSOR_NAME.name(),
+        cloudInfoProcessorName);
+  }
+
+  /**
+   * Get the CLOUD_INFO_PROCESSOR_NAME field.
+   * @return CLOUD_INFO_PROCESSOR_NAME field.
+   */
+  public String getCloudInfoProcessorName() {
+    return _record.getSimpleField(CloudConfigProperty.CLOUD_INFO_PROCESSOR_NAME.name());
+  }
+
+  /**
+   * Set the CLOUD_PROVIDER field.
+   * @param cloudProvider
+   */
+  public void setCloudProvider(CloudProvider cloudProvider) {
+    _record.setSimpleField(CloudConfigProperty.CLOUD_PROVIDER.name(), cloudProvider.name());
+  }
+
+  /**
+   * Get the CLOUD_PROVIDER field.
+   * @return CLOUD_PROVIDER field.
+   */
+  public String getCloudProvider() {
+    return _record.getSimpleField(CloudConfigProperty.CLOUD_PROVIDER.name());
+  }
+
+  public static class Builder {
+    private String _clusterName = null;
+    private CloudProvider _cloudProvider;
+    private boolean _cloudEnabled = DEFAULT_CLOUD_ENABLED;
+    private String _cloudID;
+    private List<String> _cloudInfoSources;
+    private String _cloudInfoProcessorName;
+
+    public CloudConfig build() {
+      validate();
+      return new CloudConfig(_clusterName, _cloudEnabled, _cloudProvider, _cloudID,
+          _cloudInfoSources, _cloudInfoProcessorName);
+    }
+
+    /**
+     * Default constructor
+     */
+    public Builder() {
+    }
+
+    /**
+     * Constructor with Cluster Name as input
+     * @param clusterName
+     */
+    public Builder(String clusterName) {
+      _clusterName = clusterName;
+    }
+
+    /**
+     * Constructor with CloudConfig as input
+     * @param cloudConfig
+     */
+    public Builder(CloudConfig cloudConfig) {
+      _cloudEnabled = cloudConfig.isCloudEnabled();
+      _cloudProvider = CloudProvider.valueOf(cloudConfig.getCloudProvider());
+      _cloudID = cloudConfig.getCloudID();
+      _cloudInfoSources = cloudConfig.getCloudInfoSources();
+      _cloudInfoProcessorName = cloudConfig.getCloudInfoProcessorName();
+    }
+
+    public Builder setClusterName(String v) {
```

Comment:
meaningful param name, please

Reply:
It's in Ali's code. I've rebased, and it should be gone.